File: src/main/java/com/suslovila/common/block/tileEntity/TileAntiNodeWatcher.java
package com.suslovila.common.block.tileEntity;

import com.suslovila.utils.SUSVec3;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.tileentity.TileEntity;

public class TileAntiNodeWatcher extends TileEntity {
 SUSVec3 antiNodePos;
@Override
    public void markDirty() {
        super.worldObj.markBlockForUpdate(super.xCoord, super.yCoord, super.zCoord);
        super.markDirty();
    }


    public void handleInputStack(EntityPlayer player, ItemStack stack) {
//        if (stack != null && !hasStack()){
//            ItemStack copyStack = stack.copy();
//            copyStack.stackSize = 1;
//            this.stack = copyStack;
//            --stack.stackSize;
//            markDirty();
//        }
    }


}

File: src/main/java/com/suslovila/common/block/tileEntity/TileAntiNodeWatcher.kt
package com.suslovila.common.block.tileEntity

import net.minecraft.tileentity.TileEntity

class TileAntiNodeWatcher : TileEntity() {

  val lenses = listOf<Lens>(Lens(0.5), Lens(0.5), Lens(0.5))
  public val wholeLength = lenses.sumOf { it.width };

  override fun updateEntity() {
   super.updateEntity()
   //lenses.forEach{ with(it){timer = (timer + 1) % 360} }
  }

  class Lens (val width : Double, var spinningSpeed: Double = 1.0, var angle : Double = 0.0, var turningSide: Int = 0, var speedDelta : Double = 0.005) {
   val defaultSpeedDelta = speedDelta
  }
}
